README.asciidoc

```diff
@@ -606,6 +605,30 @@ EmbeddedMaven
 Some additional examples can be found in integration tests link:/maven/impl-maven-embedded/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/pom/equipped[here] and link:/maven/impl-maven-embedded/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/invoker/equipped[here].
 
 
+=== Daemon build
+In some cases, you need to run a Maven build in the background. The first case is when the build itself starts some application (eg. `mvn spring-boot:run`). The second case is when you are building a project that is too big so can do some other operation in the meantime let the Maven built run in the background.
+In these cases you can specify that the build should be used as a daemon, which means that ShrinkWrap Resolver runs the build in a new separated thread:
+[source,java]
+....
+EmbeddedMaven.forProject("path/to/pom.xml").setGoals("spring-boot:run").useAsDaemon().build();
```

Comment:
I was considering all of them. I decided for `useAsDaemon` because:
- `runAsDaemon` evoke that the command runs the build. in other words, that when you call `runAsDaemon()` then you shouldn't do anything more, which is not correct - you still can set `wait` and need to call the `build()`

- `asDaemon` I decided more longer version just because of readability. When I compare these two versions:
~~~java
EmbeddedMaven
    .forProject(cheStarterDir.getAbsolutePath() + "/pom.xml")
    .useMaven3Version("3.5.0")
    .setGoals("spring-boot:run")
    .asDaemon()
    .withWaitUntilOutputLineMathes(".*Started Application in.*", 30, TimeUnit.SECONDS)
    .build();

EmbeddedMaven
    .forProject(cheStarterDir.getAbsolutePath() + "/pom.xml")
    .useMaven3Version("3.5.0")
    .setGoals("spring-boot:run")
    .useAsDaemon()
    .withWaitUntilOutputLineMathes(".*Started Application in.*", 30, TimeUnit.SECONDS)
    .build();
~~~
then the second one seems to be more readable (for me).